src/modules/mavlink/mavlink_main.cpp

```diff
@@ -846,55 +870,9 @@ Mavlink::send_packet()
 void
 Mavlink::send_bytes(const uint8_t *buf, unsigned packet_len)
 {
-	/* If the wait until transmit flag is on, only transmit after we've received messages.
-	   Otherwise, transmit all the time. */
-	if (!should_transmit()) {
-		return;
-	}
-
-	_last_write_try_time = hrt_absolute_time();
-
-	if (_mavlink_start_time == 0) {
-		_mavlink_start_time = _last_write_try_time;
-	}
-
-	if (get_protocol() == SERIAL) {
-		/* check if there is space in the buffer, let it overflow else */
-		unsigned buf_free = get_free_tx_buf();
-
-		if (buf_free < packet_len) {
-			/* not enough space in buffer to send */
-			count_txerrbytes(packet_len);
-			return;
-		}
-	}
-
-	size_t ret = -1;
-
-	/* send message to UART */
-	if (get_protocol() == SERIAL) {
-		ret = ::write(_uart_fd, buf, packet_len);
-	}
-
-#if defined(CONFIG_NET) || defined(__PX4_POSIX)
-
-	else {
-		if (_network_buf_len + packet_len < sizeof(_network_buf) / sizeof(_network_buf[0])) {
-			memcpy(&_network_buf[_network_buf_len], buf, packet_len);
-			_network_buf_len += packet_len;
-
-			ret = packet_len;
-		}
-	}
-
-#endif
-
-	if (ret != (size_t) packet_len) {
-		count_txerrbytes(packet_len);
-
-	} else {
-		_last_write_success_time = _last_write_try_time;
-		count_txbytes(packet_len);
+	if (_buf_len + packet_len < sizeof(_buf)) {
+		memcpy(&_buf[_buf_len], buf, packet_len);
+		_buf_len += packet_len;
```

Comment:
What happens `else`? Shouldn't we add the bytes to the `tx_err` if we can't copy it in?

Reply:
I don't think there's really a mechanism to handle it.
 - errors aren't handled mavlink side. https://github.com/PX4/Firmware/pull/12967#issuecomment-532250540
 - tx_err is used to throttle the connection, so wouldn't really help

The only real reason we'd hit this error is if the buffer we control is the wrong size or if there's an error in the mavlink library. We could do something like put a perf counter in `else` or if we had better HIL/SITL testing we could have builds that assert.

Reply:
I've added a perf counter for now.